router/peers.go

```diff
@@ -298,48 +304,19 @@ func decodePeerNoConns(dec *gob.Decoder) (nameByte []byte, nickName string, uid
 	if err = dec.Decode(&version); err != nil {
 		return
 	}
-	if err = dec.Decode(&conns); err != nil {
+	if err = dec.Decode(&connViews); err != nil {
 		return
 	}
 	return
 }
 
-func connsIterator(input []byte, fun func([]byte, string, bool, bool)) error {
-	buf := new(bytes.Buffer)
-	buf.Write(input)
-	dec := gob.NewDecoder(buf)
-	for {
-		var nameByte []byte
-		if err := dec.Decode(&nameByte); err != nil {
-			return err
-		}
-		var foundAt string
-		if err := dec.Decode(&foundAt); err != nil {
-			return err
-		}
-		var outbound bool
-		if err := dec.Decode(&outbound); err != nil {
-			return err
-		}
-		var established bool
-		if err := dec.Decode(&established); err != nil {
-			return err
-		}
-		fun(nameByte, foundAt, outbound, established)
-	}
-}
-
-func readConnsMap(peer *Peer, buf []byte, table map[PeerName]*Peer) map[PeerName]Connection {
+func readConnsMap(peer *Peer, connViews []ConnectionView, table map[PeerName]*Peer) map[PeerName]Connection {
```

Comment:
function name needs updating; we are not really 'reading' anything anymore here.